### ISSUE TITLE
MacOS Sonoma - can't use PDFwriter to print fillable PDF form from Acrobat

### ISSUE THREAD

**Issue by gittalicious** [rodyager/RWTS-PDFwriter#42]
Since updating to Sonoma, any attempt to print a fillable PDF form from Acrobat using PDFwriter results in an error: "The document could not be printed". Hitting "OK" then brings up another error: "There were no pages selected to print". The same form has worked perfectly with PDFwriter for many previous MacOS versions, and there have been no changes to the PDF form and no updates installed for Acrobat. The only thing that has changed is the update to Sonoma. 

Any solutions for this? (Have already tried rebooting, uninstalling/reinstalling PDFwriter).

**Comment by rodyager** [comment]
It appears that changes to the MacOS printing system in Sonoma have broken PDFwriter.

PDFwriter relies on features of the CUPS printing system that Apple has long indicated would not be supported in some future OS. It appears that date has arrived.

While I shall endeavour to find an alternative approach, it isn’t clear that this will be possible.

Rod

Sent from my iPhone

On 30 Sep 2023, at 9:50 am, gittalicious ***@***.***> wrote:

﻿

Since updating to Sonoma, any attempt to print a fillable PDF form from Acrobat using PDFwriter results in an error: "The document could not be printed". Hitting "OK" then brings up another error: "There were no pages selected to print". The same form has worked perfectly with PDFwriter for many previous MacOS versions, and there have been no changes to the PDF form and no updates installed for Acrobat. The only thing that has changed is the update to Sonoma.

Any solutions for this? (Have already tried rebooting, uninstalling/reinstalling PDFwriter).

**Comment by gittalicious** [comment]
Thanks for the quick response. I should mention that printing using PDFwriter from pdf's in Acrobat that are not fillable forms seems to work fine, and printing other documents (e.g., Word, emails, etc) using PDFwriter also works. The problem seems to be limited to using PDFwriter to print fillable forms from Acribat, as far as I can tell.

**Comment by rodyager** [comment]
I haven’t made the jump to Sonoma myself yet (a mission critical piece of software is still incompatible) so I’m currently relying on reports.

Once I make the jump myself, I’ll be in a better position to assess what is happening and whether there are work arounds

Rod
Sent from my iPhone

On 30 Sep 2023, at 10:25 am, gittalicious ***@***.***> wrote:

﻿

Thanks for the quick response. I should mention that printing using PDFwriter from pdf's in Acrobat that are not fillable forms seems to work fine, and printing other documents (e.g., Word, emails, etc) using PDFwriter also works. The problem seems to be limited to using PDFwriter to print fillable forms from Acribat, as far as I can tell.

**Comment by rje3** [comment]
From the MacOS release notes:

### Printing
**Deprecations**

- macOS has removed the functionality for converting PostScript and EPS files to PDF format. As a result, CoreGraphics’ CGPSConverter returns an error when invoked, ImageIO no longer converts EPS files, NSEPSImageRep does not display EPS files, and PMPrinterPrintWithFile does not accept a PostScript file for non-PostScript print queues. (110019863)

https://developer.apple.com/documentation/macos-release-notes/macos-14-release-notes#Printing

**Comment by gittalicious** [comment]
@rje3 - Is the change you posted the direct cause of the difference in functionality I observed? If so, is there a workaround I might try on my end?

**Comment by rje3** [comment]
> @rje3 - Is the change you posted the direct cause of the difference in functionality I observed? If so, is there a workaround I might try on my end?

Sorry I don’t know and looking for a solution myself. The only option that worked for me was to roll back to Ventura.

**Comment by gittalicious** [comment]
I came across this https://www.ghostscript.com/index.html - not sure if this is something that might point a way to solving this issue?

**Comment by rodyager** [comment]
It is possible that this is the cause, and I suspect it might be the cause as I have other reports telling me that PDFwriter is working with Sonoma.

The way printing works in macOS is that an application sends print data to the printing system (CUPS) in one of a number of formats (raster, pdf, postscript, eps) and CUPS manages the translation into a format that the printer understands using one of its built in translators or a translater provided by the printer driver if the printer requires a specialised format. If necessary, it will chain together several of these to get to the required format.

The notice tells us that CUPS will no longer natively translate postscript into anything else.

PDFwriter itself does no translation. It simply tells CUPS that it only accepts PDF and relies on CUPS to use its built in translators to convert from whatever the application provided to PDF. It then manages saving that PDF data stream as a file.

So it’s possible (but not definitively established) that Wine is providing data to the printing system in postscript format which CUPS can no longer translate into PDF.

If that is the case, the fix will need to come from the Wine project providing its print output in a non-postscript format.  And it will probably be a high priority for them, as, if this is the case, no printing will work on macOS for any device that doesn’t speak postscript (read most modern printers) until they fix it.

Rod




Sent from my iPhone

On 30 Sep 2023, at 10:15 pm, gittalicious ***@***.***> wrote:

﻿

I came across this https://www.ghostscript.com/index.html - not sure if this is something that might point a way to solving this issue?

**Comment by gittalicious** [comment]
I wanted to post about a workaround I discovered for my particular situation (using PDFwriter to print a fillable PDF form from Acrobat in MacOS Sonoma), in case this helps anyone else. As far as I can tell, this only needs to be done once, and then the change will persist going forward.

In the print dialogue in Acrobat, next to where it lists the printers in a drop down menu, click the button marked Advanced. Then check the box for Print as Image, and then OK to return to the print dialogue, and proceed as usual. This appears to print out a "flattened" PDF with the fillable form gone, which is the behavior I am seeking. 

I am not sure if there are any unintended consequences related to enabling this setting. Perhaps someone with more expertise can weigh in on that?

**Comment by rodyager** [comment]
Yes - I can confirm that Acrobat sends filled pdf forms to CUPS in application/postscript format, and this is the cause of the failure. Preview (which has a much more limited filled forms functionality) sends these forms to CUPS in application/pdf format, and so will "work."  Its reasonably likely that Adobe will quickly update Acrobat for Sonoma to fix this as it will affect printing filled forms to many modern printers.

**Comment by mjbaldwin** [comment]
I have been encountering the same issue -- need to save a fillable form to a "regular" PDF on macOS Sonoma which Acrobat doesn't allow, and I installed PDFWriter so it would work. I was using an older version of Acrobat Reader from 2022 (I'd disabled auto-updates), and it wouldn't work (as described in the original message of this issue), and I tried checking the "Print as image" as suggested by @gittalicious and then it works.

Intriuged, I updated Acrobat Reader to its current 2024 version to try it again, and when I click "Advanced" next to selecting "PDFwriter", every single option is grayed out, "Print As Image" is checked, and there's a non-grayed-out message "Only Print As Image is supported on MacOS14 Beta.":

<img width="536" alt="image" src="https://github.com/rodyager/RWTS-PDFwriter/assets/160959/e205db6a-81cc-4e46-b9c6-80e6da5045e0">

I'm not sure if this is what @rodyager was referring to in thinking it was likely that Adobe would update Acrobat to fix this? It's unclear to me whether this message is being produced by PDFwriter or by Adobe.

It's not perfect in that the generated PDF is a bitmap, but I just need this for basic printing and archival purposes so it's good enough for me! I'm very happy this utility exists in order to get around Adobe Reader refusing to "print" complex form PDF's to simple PDF's.
